packages/ui/src/views/repo/repo-commits/components/commits-list.tsx

```diff
@@ -1,72 +1,68 @@
-import { useMemo } from 'react'
+import { FC, useMemo } from 'react'
 
-import { Avatar, AvatarFallback, CommitCopyActions, NodeGroup, StackedList, Text } from '@/components'
+import { Avatar, AvatarFallback, CommitCopyActions, NodeGroup, StackedList } from '@/components'
 import { formatDate, getInitials } from '@/utils/utils'
+import { TypesCommit } from '@/views'
 
-import { TypesCommit } from '../types'
+type CommitsGroupedByDate = Record<string, TypesCommit[]>
 
 interface CommitProps {
   data?: TypesCommit[]
 }
 
-export const CommitsList = ({ ...props }: CommitProps) => {
-  const data = props.data
-  const commitsGroupedByDate: Record<string, TypesCommit[]> = useMemo(
-    () =>
-      data?.reduce(
-        (group, commit) => {
-          const date = formatDate(commit.committer?.when as string)
+export const CommitsList: FC<CommitProps> = ({ data }) => {
+  const entries = useMemo(() => {
+    const commitsGroupedByDate = !data
+      ? {}
+      : data.reduce<CommitsGroupedByDate>((group, commit) => {
+          const date = formatDate(commit.committer?.when ?? '')
           group[date] = (group[date] || []).concat(commit)
           return group
-        },
-        {} as Record<string, TypesCommit[]>
-      ) || {},
-    [data]
-  )
+        }, {})
+
+    return Object.entries(commitsGroupedByDate) as [string, TypesCommit[]][]
+  }, [data])
 
-  const entries = Object.entries(commitsGroupedByDate)
   const totalNodes = entries.length
 
   return (
-    <>
+    <div>
       {entries.map(([date, commitData], node_idx) => (
-        <NodeGroup.Root key={date}>
+        <NodeGroup.Root
+          className="grid-cols-[4px_1fr] gap-x-[22px] gap-y-3.5 pb-6 last:pb-0"
+          key={`${date}-${node_idx}`}
```

Comment:
Is the date not unique?

Reply:
I somehow caught this case during testing, so I added the index as well.

Reply:
Ohh... I get it, date might be an empty string (should be impossible in practice, but `when` is optional in the types 🤔). I don't think React cares if the `key` is blank, as long as it's unique and exists